src/ImageSharp/Common/Helpers/Shuffle/IComponentShuffle.cs

```diff
@@ -99,15 +99,15 @@ public void RunFallbackShuffle(ReadOnlySpan<byte> source, Span<byte> dest)
         {
             ref uint sBase = ref Unsafe.As<byte, uint>(ref MemoryMarshal.GetReference(source));
             ref uint dBase = ref Unsafe.As<byte, uint>(ref MemoryMarshal.GetReference(dest));
-            nuint n = (nuint)(uint)source.Length / 4;
+            nint n = (nint)source.Length / 4;
```

Comment:
```suggestion
            nuint n = (nuint)(uint)source.Length / 4;
```
See other comments.
This way, the JIT emits a bit-hack (a shit) instead of the costly integer division.

It's a bit tricky, but the JIT has to conform correctness, so we need to help the JIT a bit.

Reply:
We're back to using `int`'s now, so this comment is no longer applicable. Thanks for the suggestions!